datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java

```diff
@@ -0,0 +1,98 @@
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    @Inject
+    private SdxReactorFlowManager sdxReactorFlowManager;
+
+    @Inject
+    private Flow2Handler flow2Handler;
+
+    @Inject
+    private FlowChainLogService flowChainLogService;
+
+    public SdxRecoverableResponse validateRecovery(SdxCluster sdxCluster) {
+        FlowLog flowLog = flow2Handler.getFirstStateLogfromLatestFlow(sdxCluster.getId());
+        if (entitlementService.isDatalakeResizeRecoveryEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
+            if (!DatalakeResizeFlowEventChainFactory.class.getSimpleName().equals(flowChainLogService.getFlowChainType(flowLog.getFlowChainId()))) {
+                return new SdxRecoverableResponse("Last flow not part of Resize", RecoveryStatus.NON_RECOVERABLE);
+            }
+        } else {
+            return new SdxRecoverableResponse("Resize Recovery entitlement not enabled", RecoveryStatus.NON_RECOVERABLE);
+        }
+        SdxStatusEntity actualStatusForSdx = sdxStatusService.getActualStatusForSdx(sdxCluster);
+        switch (actualStatusForSdx.getStatus()) {
+            case STOP_FAILED:
+                return new SdxRecoverableResponse("Resize can be recovered from a failed stop", RecoveryStatus.RECOVERABLE);
+            case PROVISIONING_FAILED:
+                return new SdxRecoverableResponse("Failed to provision, recovery will restart old datalake, and delete the new one",
+                        RecoveryStatus.RECOVERABLE);
+            case DATALAKE_RESTORE_FAILED:
+                return new SdxRecoverableResponse("Failed to restore backup to new datalake, recovery will restart old datalake, and delete the new one",
+                        RecoveryStatus.RECOVERABLE);
+            case DELETE_FAILED:
+                return new SdxRecoverableResponse("Failed to delete old datalake, not a recoverable error", RecoveryStatus.NON_RECOVERABLE);
+
+            default:
+                return new SdxRecoverableResponse("Resize can not be recovered from this point", RecoveryStatus.NON_RECOVERABLE);
+        }
+    }
+
+    public SdxRecoveryResponse triggerRecovery(SdxCluster sdxCluster, SdxRecoveryRequest sdxRecoveryRequest) {
+        SdxStatusEntity actualStatusForSdx = sdxStatusService.getActualStatusForSdx(sdxCluster);
+        if (entitlementService.isDatalakeResizeRecoveryEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
+            switch (actualStatusForSdx.getStatus()) {
+                case STOP_FAILED:
+                    return new SdxRecoveryResponse(sdxReactorFlowManager.triggerSdxStartFlow(sdxCluster));
+                case PROVISIONING_FAILED:
+                case DATALAKE_RESTORE_FAILED:
+                default:
+                    throw new NotImplementedException("Cluster is currently in an unrecoverable state");
+            }
+        } else {
+            throw new BadRequestException("Entitlement for resize recovery is missing");
```

Comment:
there are two different messages to indicate there is no entitlement, they could be uniform.

Reply:
One is an error message, while other is part of a valid response. Since these strings are likely to be a part of any escalation that a customer has around this, I'd rather keep them unique to help a future debugger understand what a customer is seeing

Reply:
One of those is a response object, the other is an exception, so they can be searched in ENGESC-s as such, I don't like duplication, but don't insist on this.